l1-contracts/test/foundry/unit/concrete/state-transition/chain-deps/facets/Mailbox/BaseMailboxTests.t.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.24;
+
+import {MailboxTest} from "./_Mailbox_Shared.t.sol";
+import {FeeParams, PubdataPricingMode} from "contracts/state-transition/chain-deps/ZkSyncHyperchainStorage.sol";
+import {REQUIRED_L2_GAS_PRICE_PER_PUBDATA} from "contracts/common/Config.sol";
+import {MailboxFacet} from "contracts/state-transition/chain-deps/facets/Mailbox.sol";
+
+contract MailboxBaseTests is MailboxTest {
+    function setUp() public virtual {
+        prepare();
+        utilsFacet.util_setBaseTokenGasPriceMultiplierDenominator(1);
+        utilsFacet.util_setBaseTokenGasPriceMultiplierNominator(1);
+    }
+
+    function test_mailboxConstructor() public {
+        MailboxFacet m = new MailboxFacet(eraChainId);
+        assertEq(m.ERA_CHAIN_ID(), eraChainId);
+    }
+
+    function test_RevertWhen_badDenominatorInL2TransactionBaseCost() public {
+        utilsFacet.util_setBaseTokenGasPriceMultiplierDenominator(0);
+        vm.expectRevert("Mailbox: baseTokenGasPriceDenominator not set");
+        mailboxFacet.l2TransactionBaseCost(100, 10000, REQUIRED_L2_GAS_PRICE_PER_PUBDATA);
+    }
+
+    function test_successful_getL2TransactionBaseCostPricingModeValidium() public {
+        uint256 gasPrice = 10000000;
+        uint256 l2GasLimit = 1000000;
+        uint256 l2GasPerPubdataByteLimit = REQUIRED_L2_GAS_PRICE_PER_PUBDATA;
+
+        FeeParams memory feeParams = FeeParams({
+            pubdataPricingMode: PubdataPricingMode.Validium,
+            batchOverheadL1Gas: 1000000,
+            maxPubdataPerBatch: 120000,
+            maxL2GasPerBatch: 80000000,
+            priorityTxMaxPubdata: 99000,
+            minimalL2GasPrice: 250000000
+        });
+
+        utilsFacet.util_setFeeParams(feeParams);
+
+        // this was get from running the function, but more reasonable would be to
+        // have some invariants that the calculation should keep for min required gas
+        // price and also gas limit
+        uint256 l2TransactionBaseCost = 250125000000000;
```

Comment:
Ideally we would want to test this function with as many values as we can in the expected range. For example, if we fuzz all the random values used here, and that sets maxL2GasPerBatch as 0, that is definitely not true.

By the time, you get the expected ranges of each of the values in the `feeParams` why don't we write this test for 2-3 more different values of `feeParams` with the nominator and denominator value fuzzed via function parameter.

Reply:
I was thinking more about "regression testing" or using some set of predefined parameters based on construction of this function, if it's just a calculation based function. Can you explain what do you mean by getting expected ranges of values and how would we know if they are expected?